shelley/design-spec/delegation_design_spec.tex

```diff
@@ -2752,110 +2864,66 @@ \subsubsection{Pool Rewards}
-  \(N_j > \mean{N_j}\).
-  The rational behaviour for those pools would be to switch off their
-  nodes once they have successfully produced \(\mean{N_j}\)
-  blocks, leading to a period of slow chain growth at the end of each
-  epoch. This would be a hazard for the system.
-\end{itemize}
+\item
+  Even when a pool's true performance is \(1\), its actual rewards might be less
+  than its optimal rewards, because of the randomness in \(r_e\). Likewise,
+  sometimes a pool's actual rewards will be \emph{higher} than their optimal
+  rewards, if they are lucky in the leader election process.
 
-\todo{We might switch to determining the rewards directly by the number of
-  blocks produced, instead of the stake. The document will be updated once we
-  have decided on this.}
-We can lessen the first issue, and eliminate the second (which is the
-more dangerous), by using a moving average of \(\mean{p_j}\).
-We will use the exponential moving average
-\[
-  \ema{x}{e} =
-  \begin{cases}
-    x_{e_0} & e = e_0\\
-    \alpha x_e + (1-\alpha) \ema{x}{e-1} & e > e_0
-  \end{cases}
-\]
-where \(\alpha\) is a decay constant determining the relative influence of
-recent epochs, \(e\) is the current epoch, \(x_e\) is the value of \(x\) in
-epoch \(e\), and \(e_0\) is the epoch in which we start taking the average (in
-the case of the performance of stake pools, this will be the epoch in which the
-pool was created).
-
-The actual rewards for pool \(j\) (with relative stake \(\sigma_j\) and relative
-leader-stake \(s_j\)) in epoch \(e\) are then computed as
-\[
-    \hat{f}_j := \ema{\left(\frac{n_j}{\max(\mean{N_j}, 1)}\right)}{e}^\gamma\cdot f(s_j,\sigma_j).
-\]
-The difference \(f(s_j,\sigma_j)-\hat{f}_j\) will be sent to the
-treasury. In particular, this means that \emph{no pool can increase its
-own rewards by somehow preventing another pool from producing
-blocks}. In the initial release, we will track the amount of funds in
-the treasury, but there will be no mechanism for withdrawing from the
-treasury. A proper treasury will be added in a later release.
-
-Note that the sum of all maximal pool rewards never exceeds the size of the
-rewards pool, \(\sum_jf(s_j,\sigma_j)\leq R\), and that the difference
-\(R-\sum_jf(s_j,\sigma_j)\) will normally be strictly positive. This difference
-will be added to the following epoch's rewards. It is safe to do so, since it
-does \emph{not} depend on the performance of any pool.
+  These effects will balance each other, so that a well performing pool will get
+  their optimal rewards \emph{on average over multiple epochs}.
+
+\item
+  We will always have
+  \[ \sum_{\text{pools}} \hat{f}(s,\sigma,\pbar) \leq R \,, \]
+  so that all actual rewards can be paid from the rewards pot\footnote{ To prove
+    this, use that \(s' \leq \sigma' \leq z_0\), and
+    \(\sum_{\text{pools}}\beta=1\).}.
+
+  The difference \(R - \sum_{\text{pools}} \hat{f}(s,\sigma,\pbar)\) will be
+  sent to the treasury.
+\end{itemize}
```

Comment:
![image](https://user-images.githubusercontent.com/5585355/55520518-5f3d4200-5685-11e9-8361-4cbffe9267a8.png)

The `∑f(s,σ,p)` represents all reward calculations, including capping pool share by `1/k`, right? Does this mean that different between the idea pool reward and capped pool reward (for over-saturated pools) is also sent to treasury?

Because I think I remember discussions that difference caused by over-saturation is carried-over to the next epoch, and difference cause by performance is sent to the treasury. But now it also seems to be missing in the spec itself 🤔 

Spec says this:

![image](https://user-images.githubusercontent.com/5585355/55521008-8d238600-5687-11e9-9657-76c5cf92dc42.png)

So it means:
1. We make total pool
2. We subtract what goes to the treasury
3. We subtract the capped pool rewards
4. The rest also goes to treasury

It sounds like there's no carry-over to the next epoch anymore in the system. If so, then what does this arrow back from total pool to rewards represent?

![image](https://user-images.githubusercontent.com/5585355/55520944-4df53500-5687-11e9-8970-4abde009b115.png)

Where does carried-over value come from, if difference between `R` and total pool rewards is always fully sent to the treasury?

Reply:
Thanks for carefully reviewing this!

The reason for the change is this: in the original scheme, we had the hierarchy that

`rewards pot >= sum of maximal rewards >= sum of actual rewards`

The first inequality is rather straightforward to prove, and the second was trivial, since each reward was multiplied by a factor `<=1`.

But now, the apparent performance can be `>1`, and there is no reason for the second inequality to hold anymore. It's not hard to come up with counterexamples: just take one pool that is oversaturated and is chosen to produce fewer blocks than expected on average, and other pools that are not saturated are chosen to produce more. In this case, the sum of the actual rewards will be larger than the sum of the optimal rewards (note that we're not calling them maximal rewards anymore).

Reply:
Ah, I think this means there is no contribution to the rewards pot anymore from the last epoch, so I should better go and correct this in 6.4.

Reply:
Ahhhhh, so pools are over-rewarded for over-performance? Yeah, I think it was not present in the previous model.

So just to clarify - if we have `K=2` and two pools: `A` with 75% stake (so over-saturated above 50%) and `B` with 25% stake (under-saturated), and we have 100 slots to assign, and each pool gets assigned 50 slots.

`A` is expected to produce 75 blocks, but produces 50, so it received `(0.5*R)*0.66` = `~0.33*R`

`B` is expected to produce 25 blocks, but produces 50, so it receives `(0.25*R)*2` = `0.5*R`

Total distributed = `0.83*R`
Leftover `0.17*R` is sent to treasury?

My question assumed that older version of the spec said something like:
1. Ideal reward for `A` would be `0.75*R` (calculated from stake share)
2. Capped reward for `A` is `(0.5*R)` (because of `K=2`)
3. Diff is `0.25*R` is sent to next epoch
4. Slashed for performance reward for `A` is `(0.66*0.5*R)` = `0.33*R`
5. Diff is `0.17*R` is sent to treasury

And for `B` calculation would imply that it *never* can get reward larger than `0.25*R`, because of its stake. But now it is implied that `B` will get more than its share of rewards because of over-performing? Am I getting it right?

Reply:
Ok, I get this case, so if some pool is over-performing - it gets piece of rewards that would otherwise be sent to next-epoch or to treasury. But my question is about the basic case: if we have `K=2`, 100 slots, and two pools: `A` at 75% stake (over) and `B` at 25% stake (under). Randomness is fair in this epoch and they get assigned expected number of slots. So:

`A` produces 75 slots, but gets `(0.5*R)` rewards, because of `K=2`

`B` produces 25 slots and gets `(0.25*R)` rewards.

There's `(0.25*R)` left-over from capping `A`. Does this gets sent to treasury in this case? Why?

Because I seem to remember it was planned to carry-over this value to next epoch.

Reply:
The point is that generally, we will no longer have

`sum of actual rewards <= sum of optimal reward`,

so if we were to send

`reward pot - sum of optimal rewards`

to the next epoch's reward pot, we might not have enough funds to pay out all rewards.

We could make a case distinction, and send

`min (sum of optimal rewards, sum of actual rewards)`

over to the next epoch, and

`max(0, (sum of optimal rewards) - (sum of actual rewards))`

to the treasury, I guess. But I don't see a compelling reason why we should do that. It would increase the rewards for everyone in the presence of oversaturated pools. We don't really anticipate having oversaturated pools, and if we do see them, we want people to act to get away from them.

Reply:
And the only way to ensure that the actual rewards are bounded by the optimal rewards would be capping the factor from the apparent performance at 1. But that would mean that we'll systematically underpay everyone.

Reply:
Understood. Thank you very much, @kantp! 🙌